lib/rubocop/cop/rspec/factory_girl/dynamic_association_declaration.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryGirl
+        # Prefer declaring associations using the `#association` method.
+        #
+        # @example
+        #   # bad
+        #   factory :post do
+        #     comments
+        #   end
+        #
+        #   # good
+        #   factory :post do
+        #     association :comments
+        #   end
+        class DynamicAssociationDeclaration < Cop
+          MSG = 'Use `association :%<name>s` to declare association.'.freeze
+
+          def_node_search :dynamic_attribute_declarations, <<-PATTERN
```

Comment:
maybe :dynamic_association_declarations will be a better name

Reply:
Whoops. Good catch @Darhazer! 🙂